### DISCUSSION TITLE
Custom Domain Limitations

### DISCUSSION THREAD

**Discussion by WhatImListnTo** [simple-login/app#1341] [Q&A]
I ordered simple-login Premium today and purchased my own custom domain through namecheap as suggested.  I completed all the configuration, and my custom domain is functional and works with simple-login.

However, I am facing an issue that I am now confused about.  I also have a subscription with Proton Mail.  Even though my custom domain is configured for use with simple-login, Proton Mail will not allow use of my custom domain even though I do own it.  The reason is that Proton Mail requires its own custom domain configuration and will not use what was configured with simple-login.  I am super confused on why this is the case.  When I read about simple-login on its website, I see the following text:

> _SimpleLogin is the product of SimpleLogin SAS, registered in France under the SIREN number 884302134. SimpleLogin SAS is part of [Proton AG](https://proton.me/)._

Based on the text above, simple-login is a part of Proton.  With this being the case, why can't Proton and simple-login share the same custom domain configuration?  As I wrote, I am a premium subscriber and paying money for this service.  I can certainly understand not sharing it for non-premium, but I am paying for the service.  If I use my custom domain for simple-login I then cannot use it at all for Proton Mail.  This does not make lots of sense to me.

**Comment by an0maly** [comment]
I ran into this issue when I first set up SimpleLogin.  It took me awhile to figure out how to get it to work with my own domains, which are tied with e-mail addresses at ProtonMail.   I finally figured out what to do was to actually create a subdomain off of your own domain and use that for SimpleLogin.   

This does mean you need to add the corresponding DNS records, but then you can follow the directions that SimpleLogin provides, but do so for your sub-domain.   I didn't want a long e-mail address and since a lot of the point of SimpleLogin is create a random address, the sub-domain can be short.    E.g., e.<yourdomain>   so that you can get the github.qwe3@e.<yourdomain>, but then have SimpleLogin forward it to your <addy>@<yourdomain>  address.

Once you think in these terms, then you can have all of the mail for your new sub-domain to route through SimpleMail, and then onto your ProtonMail account.

**Comment by WhatImListnTo** [comment]
Thank you for your response.  I do appreciate it, and you do offer a work around for this issue.  I would prefer if Proton Mail and SimpleLogin simply share the same custom domain configuration.  This would provide perfect email functionality that I am looking for.

Take this example where I have an owned domain called:  customdomain.com

Currently I am able to configure SimpleLogin to use customdomain.com with DNS configuration.  However, I cannot share customdomain.com with Proton Mail for the reasons in my original post.  For receiving and replying to received mail, this is no issue whatsoever.  SimpleLogin provides logic to properly handle replies to all email sent to my mail boxes, and it will correctly set the reply to address based on my reading of its abilities.  All mail I receive and reply to will work fine.

The issue will be for brand new emails I create from Proton Mail.  My custom domain is hidden and not usable as it is for SimpleLogin.  When I create a new message, Proton Mail will only show my Proton email address and no others, since my custom domain is not configured (and cannot be due to it being configured for SimpleLogin).

Your alternative of a sub domain will provide a work-around, but my preference would be for SimpleLogin to be the entry point of all incoming email.  I would like SimpleLogin to forward mail to Proton Mail and my other mail boxes.  By creating a sub domain, new email I create from Proton Mail (which I would have to do; I cannot compose email using SimpleLogin) would expose my Proton Mail email address with its own sub domain.  In this case, my sub domain would be mail.customdomain.com.  I would rather avoid this, if possible.  This is where sharing the same domain configuration between these two email system would solve the issue perfectly.  It just needs to be done.

I did open a ticket with Proton Mail support, and it is in their queue.  I do appreciate your suggestion though, but it is not quite what I am looking for.

**Reply by nguyenkims** [reply]
A domain can only point to a single service for handling its email and as Proton and SimpleLogin don't handle emails in the same way, a domain can't point to both Proton and SimpleLogin.

**Reply by WhatImListnTo** [reply]
Are you suggesting that it is technically impossible for Proton AG to share the same custom domain configuration between SimpleLogin and Proton Mail?  I am not asking this in regards to what was currently implemented.  It cannot be done at all is what you are suggesting ?

Forcing different email addresses is not a good email solution.  It's a work-around at best.

**Reply by WhatImListnTo** [reply]
One simpler idea is to simply add SMTP support for sending email to SimpleLogin Premium.  This should be easier to implement and add the ability to send email using SimpleLogin.  I suggest adding this feature to premium only.  I would then ditch Proton Mail for sending all emails and only use SimpleLogin SMTP for sending.  This way I would send (using an email client) and receive email with one single email address.

**Reply by nguyenkims** [reply]
SL relies on the "reverse alias" system whereas Proton uses a more traditional email approach. Currently I don't see a simple and error-free way for supporting a domain on both SL and Proton. Please note that SL and Proton don't use the same code.

About sending emails with SMTP, for the same reason that SL relies on reverse alias for outgoing emails, the workaround wouldn't work IMO.

**Reply by WhatImListnTo** [reply]
Unfortunately, SimpleLogin combined with Proton Mail is not a good solution that will meet my needs.  I will leave it at that.

**Comment by alvintownsend** [comment]
@nguyenkims thank you for the responses, I also wish to do the same thing as @WhatImListnTo - without using a subdomain have `alias@example.com` be an alias for my ProtonMail email address of `me@example.com`.

Understanding the technical challenges, is there an appropriate place to add this as a feature request for you and Proton? I'm sure there are quite a few folks who would love to have this feature.

**Reply by nguyenkims** [reply]
I hope there's a way to do that too but unfortunately we can't think of any technical solution that would allow a domain to be handled by both Proton and SimpleLogin. You can create a feature request on this forum, maybe someone can come up with a solution ...

**Reply by alvintownsend** [reply]
Thanks for the reply!

I recognize the difficulty; you'd have to share mail servers/MX records, and figure out how to differentiate aliases from actual addresses, and probably a lot more. It'd require a great number of changes.

I'll add a suggestion here and in [Proton's UserVoice](https://protonmail.uservoice.com/forums/284483-proton-mail) system, maybe one day... 😄
